### PR TITLE
AB#70931 Checks to see if the local LTI tool is already present befor…

### PR DESCRIPTION
**Pull request by lardo-de-arnaud**
…e attempting to add it. 

Is it OK to use temporary files like this. It seemed easy to do.

I couldn't find a better way to get the list of developer keys which have been deployed.

### CODE REVIEW

File: add-LTI-dev/add-LTI-dev.sh [hunk 1]
```diff
@@ -28,15 +30,52 @@ read -r -a client_ids_array <<< "${client_ids}"
...
-  curl -X POST https://${host}/api/v1/accounts/1/external_tools  \
-    -H "Authorization: Bearer ${token}" \
-    -F "client_id=${client_id}" | jq -r '.name' | tr '\n' ' '; \
-    printf "successfully set up\n\n" \
-    || printf "Unable to set tool with client ID ${client_id}. Please do this manually.\n\n"
-    printf "\n\n"
+
+  # generate the developer_key_id, remove the 122010000... bit
+  tmp="${client_id:5}"
+  developer_key_id="$((10#$tmp))"
```
Comment (buckett) [inline add-LTI-dev/add-LTI-dev.sh:48]: What is this doing?

**Review by buckett — APPROVED** [review]
LGTM.

You can use `curl -f` if you're wanting to detect an error as then anything other than a 200 response will cause curl to exit with a failure which you can then use in the script.

`tmp=$(mktemp ${client_id}-XXXX)` is a way of generating a temp files that don't overwrite each other.

File: add-LTI-dev/add-LTI-dev.sh (hunk 1 above)
Comment (lardo-de-arnaud) [reply add-LTI-dev/add-LTI-dev.sh:48]: cunningly extracting the tool ID from the 12201000...... identifier. It lops off the first 5 characters then dumps all the leading zeroes.

**Review by buckett — APPROVED** [review]
I wasn't expecting you to make the mktemp change, just suggesting as a possibility, but nice one.

**Pull request merged**